test/e2e/specs/editor/blocks/table.spec.js

```diff
@@ -0,0 +1,276 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Table', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'displays a form for choosing the row and column count of the table', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( { name: 'core/table' } );
+
+		// Check for existence of the column count field.
+		const columnCountInput = page.locator(
+			'role=spinbutton[name="Column count"i]'
+		);
+		await expect( columnCountInput ).toBeVisible();
+
+		// Modify the column count.
+		await columnCountInput.click();
+		await expect( columnCountInput ).toHaveValue( '2' );
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.type( '5' );
+
+		// Check for existence of the row count field.
+		const rowCountInput = page.locator(
+			'role=spinbutton[name="Row count"i]'
+		);
+		await expect( rowCountInput ).toBeVisible();
+
+		// Modify the row count.
+		await rowCountInput.click();
+		await expect( rowCountInput ).toHaveValue( '2' );
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.type( '10' );
+
+		// Create the table.
+		await page.click( 'role=button[name="Create Table"i]' );
+
+		// Expect the post content to have a correctly sized table.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'allows text to by typed into cells', async ( { editor, page } ) => {
+		await editor.insertBlock( { name: 'core/table' } );
+
+		// Create the table.
+		await page.click( 'role=button[name="Create Table"i]' );
+
+		// Click the first cell and add some text.
+		await page.click( 'role=textbox[name="Body cell text"] >> nth=0' );
```

Comment:
We should append `i` to most `name`s in the role selectors since most of them are descriptive text and are not case-sensitive.

```suggestion
		await page.click( 'role=textbox[name="Body cell text"i] >> nth=0' );
```

Same applies to other selectors in this PR below ;)